### PR TITLE
Implement changes to support a custom Minio Endpoint

### PR DESCRIPTION
**Pull request by Atharex**
The changes include some pom.xml adaptations from a minio S3 java client example

The modifications support using a minio endpoint, when it is specified in the configuration

### CODE REVIEW

File: backend/pom.xml [hunk 1]
```diff
@@ -25,12 +25,26 @@
 		<liquibase.version>3.8.0</liquibase.version>
 		<prometheus.version>0.8.1</prometheus.version>
 		<protobuf.version>3.8.0</protobuf.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (ravishetye) [inline backend/pom.xml]: indentation is off

File: backend/pom.xml [hunk 2]
```diff
@@ -218,7 +232,7 @@
 	</dependencies>
 
 	<build>
-		<finalName>modeldb-${version}-client-build</finalName>
+		<finalName>modeldb-${project.version}-client-build</finalName>
```
Comment (ravishetye) [inline backend/pom.xml:221]: was this necessary?

File: backend/pom.xml [hunk 3]
```diff
@@ -25,12 +25,26 @@
...
+        <dependencies>
+            <dependency>
+                <groupId>com.amazonaws</groupId>
+                <artifactId>aws-java-sdk-bom</artifactId>
+                <version>1.11.794</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
```
Comment (ravishetye) [inline backend/pom.xml]: around line 139 we do have dependency on
```
<artifactId>aws-java-sdk-s3</artifactId>
<version>1.11.794</version>
```

A quick read on sdk-bom tells me it is useful when downloading multiple artifacts from com.amazonaws, that does not seem to be the case in modeldb. Was this change necessary?

File: backend/pom.xml [hunk 4]
```diff
@@ -249,6 +263,14 @@
 					</execution>
 				</executions>
 			</plugin>
+            <plugin>
```
Comment (ravishetye) [inline backend/pom.xml]: indentation

File: backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/S3Service.java [hunk 5]
```diff
@@ -30,18 +33,37 @@
...
+  private String keyName;
+  private String UploadFileName;
 
   public S3Service(String cloudBucketName) {
     App app = App.getInstance();
     String cloudAccessKey = app.getCloudAccessKey();
     String cloudSecretKey = app.getCloudSecretKey();
-    if (cloudAccessKey != null && cloudSecretKey != null) {
+    String minioEndpoint = app.getMinioEndpoint();
+    if (cloudAccessKey != null && cloudSecretKey != null && minioEndpoint == null) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/S3Service.java]: I think this will improve the readability

```
if (cloudAccessKey != null && cloudSecretKey != null){
 if(minioEndpoint ==null){
 ...
 } else {
  //your new code 
 }
} else { 
  //read from env
}
```

**Review by ravishetye — CHANGES_REQUESTED** [review]
Thanks for your contribution @Atharex .
I had few questions and suggestions.

File: backend/pom.xml [hunk 6]
```diff
@@ -25,12 +25,26 @@
 		<liquibase.version>3.8.0</liquibase.version>
 		<prometheus.version>0.8.1</prometheus.version>
 		<protobuf.version>3.8.0</protobuf.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <aws.sdk.version>1.11.106</aws.sdk.version>
```
Comment (ravishetye) [inline backend/pom.xml]: this is unused?

File: backend/pom.xml (hunk 1 above)
Comment (Atharex) [reply backend/pom.xml]: True, I used spaces there... Replaced them with tabs. Should be ok now

File: backend/pom.xml (hunk 2 above)
Comment (Atharex) [reply backend/pom.xml:221]: when running `mvn package -Dmaven.test.skip=true` I got this warning in the output:

```
[WARNING]
[WARNING] Some problems were encountered while building the effective model for ai.verta.modeldb:modeldb:jar:1.0-SNAPSHOT
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
```

This is the reason why I've changed it. But can also revert it, if you believe this shouldn't be included!

File: backend/pom.xml (hunk 4 above)
Comment (Atharex) [reply backend/pom.xml]: Also used spaces there... replaced them now with tabs

File: backend/pom.xml (hunk 3 above)
Comment (Atharex) [reply backend/pom.xml]: No, it was not necessary. I've tested it also without sdk-bom and it worked, so I'll remove this bit

File: backend/pom.xml (hunk 6 above)
Comment (Atharex) [reply backend/pom.xml]: seems so... I was using a prebuilt minio example from another project and inadvertently transferred over more than was needed it seems. I've also removed this now

File: backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/S3Service.java (hunk 5 above)
Comment (Atharex) [reply backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/S3Service.java]: Agreed. I've adapted the file to this structure and I've also removed a couple of variables which were not used 
```
-  private String keyName;
-  private String UploadFileName;
```

**Comment by Atharex** [comment]
Even though this adaption enables artifact storage to Minio, the artifacts can only be downloaded from Minio itself, but not from the ModelDB UI.

I guess the problem lies with the path to the file stored in the DB. Even when using a custom Minio endpoint, the path to the artifact stored in Postgres still points to Amazon s3 as the base URL
`s3://modeldb-bucket/0b4c3469bc2751f33d7f32c17ba28e92efc4eb17449f25d4283f032e38d71549/dataset.pkl`

File: backend/pom.xml (hunk 2 above)
Comment (ravishetye) [reply backend/pom.xml:221]: ok makes sense , thanks for the change and the explanation

**Comment by ravishetye** [comment]
Thanks for the analysis. Would you like to fix that in a separate PR? someone from the team can guide you with the relevant code section.

Out team could fix this as well, but if will be helpful how the URL for minio should look like.

**Review by ravishetye — APPROVED** [review]
Thanks @Atharex for the fix . This LGTM to be merged in.
